sklearn/metrics/tests/test_ranking.py

```diff
@@ -736,6 +736,18 @@ def test_roc_auc_score_multiclass_error(msg, kwargs):
         roc_auc_score(y_true, y_prob, **kwargs)
 
 
+def test_assert_ovr_roc_auc_chance_level():
+    # Build equal probability predictions to multiclass problem
+    y_true = np.array([3, 1, 2, 0])
+    y_pred = 0.25 * np.ones((4, 4))
+
+    macro_roc_auc = roc_auc_score(y_true, y_pred, multi_class="ovr", average="macro")
+    assert_allclose(macro_roc_auc, 0.5)
+
+    micro_roc_auc = roc_auc_score(y_true, y_pred, multi_class="ovr", average="micro")
+    assert_allclose(micro_roc_auc, 0.5)
+
```

Comment:
As commented above, those new assertions do not cover the `average="micro"` case at the moment.

Furthermore, please also add a new test that checks the equivalence between `average="micro"` with running the `roc_auc_score` function on the manually raveled `y_true` / `y_pred` on some imperfect predictions.

For instance, here is a simple way to get better than chance predicted probabilities on an slightly imbalanced multiclass classification problem:


```python
from scipy import stats

y_pred = stats.dirichlet.rvs([2.0, 1.0, 0.5], size=1000, random_state=0)
y_true = np.asarray(
    [stats.multinomial.rvs(n=1, p=y_pred_i, random_state=0).argmax() for y_pred_i in y_pred]
)
```

You can use the `global_random_seed` fixture instead of seeding to 0 to make the test independent to the seed.